DatadogTrace/Sources/OpenTelemetry/OTelSpanBuilder.swift

```diff
@@ -0,0 +1,140 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import OpenTelemetryApi
+
+internal class OTelSpanBuilder: OpenTelemetryApi.SpanBuilder {
+    var tracer: DatadogTracer
+    var spanName: String
+    var spanKind = SpanKind.client
+    var attributes: [String: OpenTelemetryApi.AttributeValue]
+    var startTime: Date?
+    var active: Bool
+    var parent: Parent
+
+    enum Parent {
+        case currentSpan
+        case span(OpenTelemetryApi.Span)
+        case spanContext(OpenTelemetryApi.SpanContext)
+        case noParent
+
+        func context() -> OpenTelemetryApi.SpanContext? {
+            switch self {
+            case .currentSpan:
+                return OpenTelemetry.instance.contextProvider.activeSpan?.context
+            case .span(let span):
+                return span.context
+            case .spanContext(let context):
+                return context
+            case .noParent:
+                return nil
+            }
+        }
+    }
+
+    init(
+        active: Bool,
+        attributes: [String: OpenTelemetryApi.AttributeValue],
+        parent: Parent,
+        spanKind: SpanKind,
+        spanName: String,
+        startTime: Date?,
+        tracer: DatadogTracer
+    ) {
+        self.tracer = tracer
+        self.spanName = spanName
+        self.spanKind = spanKind
+        self.attributes = attributes
+        self.startTime = startTime
+        self.active = active
+        self.parent = parent
+    }
+
+    func setParent(_ parent: OpenTelemetryApi.Span) -> Self {
+        self.parent = .span(parent)
+        return self
+    }
+
+    func setParent(_ parent: OpenTelemetryApi.SpanContext) -> Self {
+        self.parent = .spanContext(parent)
+        return self
+    }
+
+    func setNoParent() -> Self {
+        self.parent = .noParent
+        return self
+    }
+
+    // swiftlint:disable unavailable_function
+    func addLink(spanContext: OpenTelemetryApi.SpanContext) -> Self {
+        fatalError("Not implemented yet")
+    }
+
+    func addLink(spanContext: OpenTelemetryApi.SpanContext, attributes: [String: OpenTelemetryApi.AttributeValue]) -> Self {
+        fatalError("Not implemented yet")
+    }
+    // swiftlint:enable unavailable_function
+
+    func setSpanKind(spanKind: OpenTelemetryApi.SpanKind) -> Self {
+        self.spanKind = spanKind
+        return self
+    }
+
+    func setStartTime(time: Date) -> Self {
+        self.startTime = time
+        return self
+    }
+
+    func setActive(_ active: Bool) -> Self {
+        self.active = active
+        return self
+    }
+
+    func startSpan() -> OpenTelemetryApi.Span {
+        let parentContext = parent.context()
+        let traceId: TraceId
+        let spanId = SpanId.random()
```

Comment:
suggestion/ Abstracting [trace IDs generation](https://github.com/DataDog/dd-sdk-ios/blob/c91484f5c71039b8ee464e9c697707c0634f4dfe/DatadogInternal/Sources/NetworkInstrumentation/TraceID.swift#L94-L99) to separate type might be handy in some unit tests.

Reply:
this is not required at this point - so going to keep as it is. We can refactor when needed.